bitshuffle/library.py

```diff
@@ -0,0 +1,231 @@
+    warn(integer, *argv, severity=kwargs['severity'])
+    exit_with_code(integer)
+
+
+def exit_successfully():
+    '''exit, successfully'''
+    if DEBUG:
+        warn(0, 0)
+    exit_with_code(0)
+
+
+def shasum(data):
+    '''Return sha256 sum of input, encdoded in base16'''
+    return sha256(data).hexdigest()
+
+
+def encode_data(data, chunksize=2048, compresslevel=5, compress=bz2.compress):
+    """encode_data
+
+    Compress the given data (which should be bytes), chunk it into chunksize
+    sized chunks, base64 encode the chunks, and return the lot as a list of
+    chunks, which are strings.
+
+    :param data: bytes
+    :param compresstype: string: bz2 or gzip
+    :param compresslevel: int: 1-9 inclusive
+    """
+
+    data = compress(data, compresslevel=compresslevel)
+    chunks = []
+    chunkptr = 0
+    while True:
+        chunk = data[chunkptr:chunkptr + chunksize]
+        chunkptr += chunksize
+
+        chunks.append(base64.b64encode(chunk))
+
+        if chunkptr >= len(data):
+            chunk = data[chunkptr:]
+            chunks.append(base64.b64encode(chunk))
+            break
+
+    return [chunk for chunk in chunks if len(chunk) > 0]
+
+
+def encode_packet(data, seqnum, seqmax, compress=bz2.compress, msg=DEFAULT_MSG,
+                  file_hash=None):
+    """encode_packet
+
+    Take an already encoded data string and encode it to a BitShuffle data
+    packet.
+
+    :param data: bytes
+    """
+
+    encoding = "base64"
+    packet_hash = shasum(data)
+    data = data.decode()
+
+    fmt = "((<<{}|{}|{}|{}|{}|{}|{}|{}"
+    packet = fmt.format(msg, COMPATLEVEL, encoding, compress.__module__,
+                        seqnum, seqmax, packet_hash, data)
+    if file_hash is not None:
+        packet += '|'
+        packet += file_hash
+    packet += '>>))'
+    return packet
+
+
+def encode_file(fhandle=stdin, chunksize=2048, compresslevel=5,
+                compress=bz2.compress, msg=DEFAULT_MSG):
+    """encode_file
+
+    Encode the file from fhandle and return a list of strings containing
+    BitShuffle data packets.
+
+    :param fhandle:
+    """
+
+    try:
+        # Python 3
+        data = fhandle.buffer.read()
+    except AttributeError:
+        data = fhandle.read()
+    file_hash = shasum(data)
+    chunks = encode_data(data, chunksize, compresslevel, compress)
+    seqmax = len(chunks) - 1
+    seqnum = 0
+    packets = []
+    for chunk in chunks:
+        if seqnum == 0 or seqnum == seqmax:
+            packet = encode_packet(chunk, seqnum, seqmax, compress, msg,
+                                   file_hash)
+        else:
+            packet = encode_packet(chunk, seqnum, seqmax, compress, msg)
+        packets.append(packet)
+
+        seqnum += 1
+
+    return packets
+
+
+def decode(message):
+    '''Decode any number of packets previously encoded with bitshuffle
+    Assumes all packets are part of the same message'''
+    _, _, _, compression, seq_num, \
```

Comment:
This is very nitpicky, and not really related to #60, but these variables names do not make it obvious that these values are indicies. Reading `file_hash`, you would expect the variable to hold the hash, not the index thereof. 

I think it's been like this for a while, but it just occurred to me.

Reply:
added https://github.com/charlesdaniels/bitshuffle/issues/63